services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/util/DeviceRegistryUtils.java

```diff
@@ -242,4 +242,39 @@ public static void checkCredential(
             }
         }
     }
+
+    /**
+     * Checks if the properties in the given client context matches with the extension properties of 
```

Comment:
It would be good to describe how the client context is matched against the extension properties as I do not think that this is straight forward.

Reply:
How about describing (how is being matched) in the paragraph next to the description as below?  WDYT? 

```
    /**
     * Checks if the properties in the given client context matches with the extension properties of the
     * given credential.
     * <p>
     * The matching properties in the client context and credential extensions are filtered out based 
     * on their property names. If their values are equal then {@code true} is returned. Otherwise {@code false}.
```

Reply:
we'll also need to mention the special handling of properties that only exist in the context but not in the extensions ...

Reply:
I have also added the above to the JavaDoc.